package-res/resources/web/pentaho/type/_type.js

```diff
@@ -0,0 +1,1360 @@
+       *
+       * Setting to `undefined` causes this attribute to use the inherited value,
+       * except for the root type, `Instance.type` (which has no ancestor), where this attribute is `null`.
+       *
+       * Setting to `null` or to an empty string clears the attribute and sets it to `null`,
+       * ignoring any inherited value.
+       *
+       * @type {?nonEmptyString}
+       */
+      get description() {
+        return this._description;
+      },
+
+      set description(value) {
+        if(value === undefined) {
+          this._resetDescription();
+        } else {
+          this._description = nonEmptyString(value);
+        }
+      },
+
+      _resetDescription: function() {
+        if(this !== _type) {
+          delete this._description;
+        }
+      },
+      //endregion
+
+      //region category property
+
+      // -> nonEmptyString, Optional, Inherited, Configurable, Localized
+      // "" -> null conversion
+
+      _category: null,
+
+      /**
+       * Gets or sets the category associated with this type.
+       *
+       * The category is used primarily to group similar types (or instances of) in a user interface.
+       *
+       * Attempting to set to a non-string value type implicitly
+       * converts the value to a string before assignment.
+       *
+       * Setting to `undefined` causes this attribute to use the inherited value,
+       * except for the root type, `Instance.type` (which has no ancestor), where the attribute is `null`.
+       *
+       * Setting to `null` or to an empty string clears the attribute and sets it to `null`,
+       * thus ignoring any inherited value.
+       *
+       * @type {?nonEmptyString}
+       * @see pentaho.type.Type#isBrowsable
+       * @see pentaho.type.Type#ordinal
+       */
+      get category() {
+        return this._category;
+      },
+
+      set category(value) {
+        if(value === undefined) {
+          this._resetCategory();
+        } else {
+          this._category = nonEmptyString(value);
+        }
+      },
+
+      _resetCategory: function() {
+        if(this !== _type) {
+          delete this._category;
+        }
+      },
+      //endregion
+
+      //region helpUrl property
+
+      // -> nonEmptyString, Optional, Inherited, Configurable, Localized?
+      // "" -> null conversion
+
+      _helpUrl: null,
+
+      /**
+       * Gets or sets an URL pointing to documentation associated with this type.
+       *
+       * Attempting to set to a non-string value type implicitly
+       * converts the value to a string before assignment.
+       *
+       * Setting to `undefined` causes this attribute to use the inherited value,
+       * except for the root type, `Instance.type` (which has no ancestor), where the attribute is `null`.
+       *
+       * Setting to `null` or to an empty string clears the attribute and sets it to `null`,
```

Comment:
Suggest:

Setting this to `null` or to an empty string clears the attribute and sets it to `null`,